imgui_node_editor.cpp

```diff
@@ -902,6 +926,43 @@ ImCubicBezierPoints ed::Link::GetCurve() const
     return result;
 }
 
+ImLinePoints ed::Link::GetPathSameNode() const
+{
+    ImLinePoints result;
+
+    result.P0 = m_Start + ImVec2(0, 10);
+    result.P1 = m_Start + ImVec2(0, 30);
+    result.P2 = result.P1 + ImVec2(100, 0);
+    result.P3 = ImVec2(result.P2.x, m_End.y - 50);
+    result.P4 = result.P3 - ImVec2(100, 0);
+    result.P5 = m_End - ImVec2(0, 40);
+
+    return result;
+}
+
+bool ed::Link::TestHitSameNode(const ImLinePoints& path, const ImVec2& point, float extraThickness) const
+{
+    auto distance = [](const ImVec2& p1, const ImVec2& p2) {
+        return ImSqrt(ImPow(p1.x - p2.x, 2) + ImPow(p1.y - p2.y, 2));
+    };
+    auto isInlLine = [&distance, &extraThickness](const ImVec2& l1, const ImVec2& l2, const ImVec2& p1) {
+        if (distance(l1, p1) + distance(l2, p1) <= distance(l1, l2) + extraThickness)
+            return true;
+        return false;
```

Comment:
```suggestion
        return distance(l1, p1) + distance(l2, p1) <= distance(l1, l2) + extraThickness;
```

Reply:
Assuming you want to return true if the point is no more than `extraThickness` from the line away, this is not quite accurate. It might be an ok approximation for small values for extraThickness though, since cos(alpha) ~= alpha for very small alpha. See also for example https://stackoverflow.com/questions/849211/shortest-distance-between-a-point-and-a-line-segment